File: 2418-sort-the-people/2418-sort-the-people.js
/**
 * @param {string[]} names
 * @param {number[]} heights
 * @return {string[]}
 */
var sortPeople = function(names, heights) {
        map = {};
        for (let i = 0; i < names.length; i++) {
            map[heights[i]] = names[i];
        }
        heights.sort(function(a,b){return a-b}).reverse();
        const res = [];
        for (let i = 0; i < names.length; i++) {
            res.push(map[heights[i]]);
        }
        return res;
};

File: 2418-sort-the-people/2418-sort-the-people.java
class Solution {
    public String[] sortPeople(String[] names, int[] heights) {
        Map<Integer, String> map = new HashMap<>();
        for (int i = 0; i < names.length; i++) {
            map.put(heights[i], names[i]);
        }
        Arrays.sort(heights); // sorted in ascending order
        String[] res = new String[names.length];
        for (int i = 0; i < names.length; i++) {
            res[res.length - 1 - i] = map.get(heights[i]);
        }
        return res;
    }
}